library/webpack-plugins/chunk-manifest-plugin.js

```diff
@@ -0,0 +1,45 @@
+/*
+  This plugin listens for the `chunk-asset` hook and records the following information:
+
+  {
+    'chunkName': {
+      filename: 'public/path/to/chunk',
+      hasRuntime: true/false,
+      parents: ['chunkName1', ..., 'chunkNameN']
+    }
+  }
+
+  It supplies a `chunk-manifest` hook that can be used by other plugins to obtain this information.
+
+  On top of that it generates a `chunk-manifest.json` file that can be used to render the correct
+  scripts on a page.
+*/
+
+const { RawSource } = require('webpack-sources')
+
+module.exports = chunkManifestPlugin
+
+function chunkManifestPlugin() {
+  return {
+    apply: compiler => {
+      compiler.plugin('compilation', compilation => {
+
+        const chunkAssets = {}
+
+        compilation.plugin('chunk-asset', (chunk, filename) => {
+          if (!chunk.name || filename.includes('hot-update')) return
+          chunkAssets[chunk.name] = {
+            filename,
+            hasRuntime: chunk.hasRuntime(),
+            parents: chunk.parents.map(c => c.name).filter(Boolean)
+          }
+        })
+
+        compilation.plugin('additional-chunk-assets', chunks => {
+          compilation.applyPlugins('chunk-manifest', chunkAssets)
+          compilation.assets['chunk-manifest.json'] = new RawSource(JSON.stringify(chunkAssets, null, '  '))
```

Comment:
Pro-tip: 
```js
JSON.stringifty(whatever, null, 2) // = two whitespaces

Reply:
From MDN:

> A String or Number object that's used to insert white space into the output JSON string for readability purposes. If this is a Number, it indicates the number of space characters to use as white space; this number is capped at 10 (if it is greater, the value is just 10). Values less than 1 indicate that no space should be used. If this is a String, the string (or the first 10 characters of the string, if it's longer than that) is used as white space. If this parameter is not provided (or is null), no white space is used.

I thought `  ` would be more of an indication of what it was than `2`